jolie/src/jolie/OOITBuilder.java

```diff
@@ -1670,6 +1519,47 @@ public void visit( ProvideUntilStatement n )
 		currProcess = new ProvideUntilProcess( (NDChoiceProcess)buildProcess( n.provide() ), (NDChoiceProcess)buildProcess( n.until() ) );
 	}
 
+	@Override
+	public void visit(TypeChoiceDefinition n) {
+		Type currType1;
+		Type currType2;
+		boolean backupInsideType = false;
+		insideType = true;
+
+		if ( n.untypedSubTypes() ) {
```

Comment:
For building currType1 and currType2, can't you just recursively call this visitor (OOITBuilder) ?